src/utils.zig

```diff
@@ -1875,6 +1875,18 @@ pub const BinaryReader = struct {
 		defer self.remaining = self.remaining[len + 1 ..];
 		return self.remaining[0..len :delimiter];
 	}
+
+	/// Read a slice of bytes into newly allocated memory.
+	/// Caller owns the memory.
+	pub fn readAlloc(self: *BinaryReader, allocator: NeverFailingAllocator, length: usize) error{OutOfBounds, IntOutOfBounds}![]u8 {
```

Comment:
I personally would prefer just a readSlice function, and let the caller call `allocator.dupe` on the result.